sqlmodel/main.py

```diff
@@ -585,9 +585,24 @@ def get_config(name: str) -> Any:
             setattr(new_cls, "__abstract__", True)  # noqa: B010
         return new_cls
 
-    # Override SQLAlchemy, allow both SQLAlchemy and plain Pydantic models
     def __init__(
         cls, classname: str, bases: Tuple[type, ...], dict_: Dict[str, Any], **kw: Any
+    ) -> None:
+        return cls.__do_init__(bases, dict_, kw)  # type: ignore
+
+    def sqlmodel_rebuild(cls) -> None:
+        reg = cls._sa_registry
+        # clear any exisiting mappers for the cls
+        managers = [m for m in reg._managers if m.class_ == cls]
+        for m in managers:
+            reg._dispose_manager_and_mapper(m)
+            del reg._managers[m]
+
+        return cls.__do_init__(cls.__bases__, cls.__dict__, {})  # type: ignore
```

Comment:
@samuelcolvin any advice on how to correctly invoke `__do_init__()` here so that whatever magic in inserting `classname` as an argument works the same as before?

Reply:
There was no magic. It was a stupid error in passing passing the `kwargs` without `**`